spec/game_spec.rb

```diff
@@ -13,27 +13,53 @@ def output(message)
 end
 
 class TestPresenter
-  def print_board
-    "______"
+  def print_board(board)
+    board.board
   end
 end
+
+class TestBoard
+  attr_reader :board
+  def initialize
+    @board = [1, 2, 3]
+  end
+
+  def []=(num, val)
+    @board[num] = val
+  end
+
+  def place_mark(player, position)
+    @board[position - 1] = player.mark
+  end
+end
+
+class TestPlayer
+  def mark
+    "Z"
+  end
+end
+
 describe "Game" do
-  it "passes this example test" do
-    pass = true
-    expect(pass).to eq(true)
+  before(:each) do
+    @game = Game.new(TestDisplay.new, TestPresenter.new, TestBoard.new)
   end
 
   it "displays the welcome message on game start" do
-    game = Game.new(TestDisplay.new, TestPresenter.new)
+    @game.start
 
-    game.start
-
-    expect(game.display.state[0]).to eq("Welcome to Tic Tac Toe!")
+    expect(@game.display.state[0]).to eq("Welcome to Tic Tac Toe!")
   end
 
   it "it displays the board from the presenter" do
-    game = Game.new(TestDisplay.new, TestPresenter.new)
-    game.play
-    expect(game.display.state[0]).to eq("______")
+    @game.print_board
+    expect(@game.display.state[0]).to eq([1, 2, 3])
+  end
+
+  it "Allows a player to place their mark on the board" do
+    player = TestPlayer.new
+
+    @game.board.place_mark(player, 1)
+    @game.print_board
+    expect(@game.display.state[0][0]).to eq("Z")
   end
```

Comment:
This is an interesting test in that you don't actually call @game.play but test a couple of methods within play.  Do you think this is a good way to test?  We want to test our public facing methods, for example @game.play.  @game.print_board should be a private method, as it's really never used outside of game.  Really the second test for game is it "plays the game" and you could show the sequence of boards displayed.  
But this hits on the problem with "newing up" a model in initialize - it's very difficult to unit test.  If you passed in an instance of player to game, as you do with display, presenter, and board, then you could easily test the play method.

Reply:
Will have to think about this one, it's the largest change to implement. I get what you mean about the unit testing, I was getting worried that game was starting to take in too many arguments.

Reply:
It's okay to initialize the game with a player as you do with display, presenter, and board.  When you have two players, you could find a way to limit the arguments (for example, game could be initialized with an array of players).

Reply:
I started writing out a test for game.play, but then stopped bc I wasn't sure what the test should be. This isn't the final version of game.play, and I'd have to alter the test to accommodate a second player anyway. That's why I wrote tests for the functions within game.play, because those smaller functions are still going to be used later. 

The test I wrote ended up failing, and I'm still not entirely sure why either. It was pushing [X,2,3] into the display.state before and after the get_input prompt.